### PR TITLE
The app should take the user back to the profile page when the user clicks the back button in the web browser

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

When viewing the profile photo, the app should take the user back to the profile page when the user clicks the back button in the web browser
### Fixed Issues

$ https://github.com/Expensify/App/issues/27856
PROPOSAL: https://github.com/Expensify/App/issues/27856#issuecomment-1727633670

### Tests

- Profile avatar page
1. Go to settings > profile
2. Click on avatar and add a photo if you don't have any
3. Click on the avatar again
4. Click on View photo
5. Click on the system/browser back button
6. Verify that the app takes the user back to the setting profile page
- Report avatar page
1. Open the workspace chat
2. Click on header to open detail page
3. Click on the avatar of  the workspace
4. Click on the system/browser back button
5. Verify that the app takes the user back to the detail page
- Workspace avatar page
1. Go to settings> workspace > select a workspace > setting
2. Add a custom avatar for workspace
3. Click on  the avatar > View photo
4. Click on the system/browser back button
5. Verify that the app takes the user back to the setting workspace page

- [x] Verify that no errors appear in the JS console

### QA Steps

- Profile avatar page
1. Go to settings > profile
2. Click on avatar and add a photo if you don't have any
3. Click on the avatar again
4. Click on View photo
5. Click on the system/browser back button
6. Verify that the app takes the user back to the setting profile page
- Report avatar page
1. Open the workspace chat
2. Click on the header to open detail page
3. Click on the avatar of  the workspace
4. Click on the system/browser back button
5. Verify that the app takes the user back to the detail page
- Workspace avatar page
1. Go to settings> workspace > select a workspace > setting
2. Add a custom avatar for workspace
3. Click on  the avatar > View photo
4. Click on the system/browser back button
5. Verify that the app takes the user back to the setting workspace page

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/129500732/91e9b3b0-f17b-4d64-913c-0ecf1f6d113d

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/129500732/20ee0799-8aa2-4a6b-bbba-c1c2451018cc

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/129500732/f018e1b3-82b1-44d9-9e37-c1f26bc6e5a1

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/129500732/152b39ac-dece-4ffc-a10a-baf21b15622f

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/129500732/e90a5db0-96c8-417c-90b8-f8f6c063a13d

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/129500732/ecf5c95b-1f89-4e7c-b5f2-de62bbbc6cb5

</details>

### CODE REVIEW

**Review by aimane-chnaif — COMMENTED** [review]
Also apply to other user profile avatars, workspace avatars

https://github.com/Expensify/App/assets/96077027/113c30a1-340d-43b2-96bd-6970fcc98ac7

**Comment by johnmlee101** [comment]
> Create a unique modal route for all image modals. This could apply to any new image modal coming in the future
Please let me know which options that you prefer

Can you elaborate on what these new routes would look like?

**Comment by johnmlee101** [comment]
Yeah separate routes seems fine for now 👍

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]
After pull main, lint will fail because all new files should be written in TS

File: src/pages/workspace/WorkspaceAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+type PolicyRoute = RouteProp<{params: {policyID: string}}>;
+
+type WorkspaceAvatarOnyxProps = {
+    policy: OnyxEntry<Policy>;
+    isLoadingApp: OnyxEntry<boolean>;
+};
+
+type WorkspaceAvatarProps = WorkspaceAvatarOnyxProps & {
+    route: PolicyRoute;
+};
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspaceAvatar.tsx:22]: Use StackScreenProps type instead, example here:
```ts
type ReportAttachmentsProps = StackScreenProps<AuthScreensParamList, typeof SCREENS.REPORT_ATTACHMENTS>;
```

Same for other files

**Review by blazejkustra — CHANGES_REQUESTED** [review]
TS-wise overall good job 🙌 

Left some minor comments

**Review by blazejkustra — APPROVED** [review]

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

**Review by aimane-chnaif — APPROVED** [review]

**Review by johnmlee101 — APPROVED** [review]
Okay tested it out a bit, it seems good 👍

File: src/pages/settings/Profile/ProfileAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        if (!ValidationUtils.isValidAccountRoute(Number(accountID)) ?? !!avatarURL) {
+            return;
+        }
+        PersonalDetails.openPublicProfilePage(accountID);
+    }, [accountID, avatarURL]);
+
+    return (
+        <AttachmentModal
+            // @ts-expect-error TODO: Remove this once AttachmentModal (https://github.com/Expensify/App/issues/25130) is migrated to TypeScript.
+            headerTitle={personalDetail?.displayName ?? ''}
```
Comment (mollfpr) [inline src/pages/settings/Profile/ProfileAvatar.tsx:38]: ✋ Coming from https://github.com/Expensify/App/issues/35514

The header title here will show empty on a hidden user. So we should use `PersonalDetailsUtils.getDisplayNameOrDefault` to get the correct display name for such a condition.

File: src/pages/ReportAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    isLoadingApp: OnyxEntry<boolean>;
+    policies: OnyxCollection<Policy>;
+};
+
+type ReportAvatarProps = ReportAvatarOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.REPORT_AVATAR>;
+
+function ReportAvatar({report = {} as Report, policies, isLoadingApp = true}: ReportAvatarProps) {
+    const policy = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID ?? '0'}`];
+    const isArchivedRoom = ReportUtils.isArchivedRoom(report);
+    const policyName = isArchivedRoom ? report?.oldPolicyName : policy?.name;
```
Comment (allroundexperts) [inline src/pages/ReportAvatar.tsx:26]: We should have used `ReportUtils.getPolicyName(report, false, policy)` to get the policy name. This caused https://github.com/Expensify/App/issues/35262

**Pull request merged**